main/modes/quickSettings/quickSettings.c

```diff
@@ -0,0 +1,337 @@
+/**
+ * @file quickSettings.c
+ * @author dylwhich (dylan@whichard.com)
+ * @brief A Quick Settings menu mode to be drawn on top of other apps
+ * @date 2023-08-14
+ *
+ */
+
+//==============================================================================
+// Includes
+//==============================================================================
+
+#include "quickSettings.h"
+#include "menu.h"
+#include "menu_utils.h"
+#include "menuQuickSettingsRenderer.h"
+
+#include "settingsManager.h"
+
+#include <stdint.h>
+#include <stdbool.h>
+
+//==============================================================================
+// Defines
+//==============================================================================
+
+//==============================================================================
+// Enums
+//==============================================================================
+
+//==============================================================================
+// Structs
+//==============================================================================
+
+typedef struct
+{
+    menu_t* menu;                          ///< The menu structure
+    menuQuickSettingsRenderer_t* renderer; ///< Renderer for the menu
+    font_t font;                           ///< The font used for menu text
+
+    wsg_t iconGeneric;
```

Comment:
What is `iconGeneric` used for?

Reply:
I wanted to have a default case and I was at one point going to add a few more settings and didn't want to come up with icons for those too. I can get rid of it if it doesn't end up being used.

Reply:
Ah, that makes sense. I'm always on the lookout for unused assets before bloat becomes a problem (just ask @Brycey92)